From f87dad3742c649dd6a4a2d71f23a8ee2d6976469 Mon Sep 17 00:00:00 2001
From: Pavel Maloletkov <pavllick@gmail.com>
Date: Fri, 6 Jun 2025 15:53:52 -0400
Subject: [PATCH] drivers: net: http: fix __z_http_service_define fields
 incorrect order

Fixes an error:
error: designator order for field 'http_service_desc::backlog' does not match declaration order in 'const http_service_desc'
When compiled for C++

Signed-off-by: Pavel Maloletkov <pavllick@gmail.com>
---
 include/zephyr/net/http/service.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/zephyr/net/http/service.h b/include/zephyr/net/http/service.h
index 20e4d30a995c..b43de0c73dbf 100644
--- a/include/zephyr/net/http/service.h
+++ b/include/zephyr/net/http/service.h
@@ -99,8 +99,8 @@ struct http_service_desc {
 		.fd = &_name##_fd,                                                                 \
 		.detail = (void *)(_detail),                                                       \
 		.concurrent = (_concurrent),                                                       \
-		.data = &_name##_data,                                                             \
 		.backlog = (_backlog),                                                             \
+		.data = &_name##_data,                                                             \
 		.res_begin = (_res_begin),                                                         \
 		.res_end = (_res_end),                                                             \
 		.res_fallback = (_res_fallback),                                                   \
